.github/workflows/ci.yml

```diff
@@ -0,0 +1,127 @@
+name: ci
```

Comment:
Why are all these jobs in a single workflow? Do you have to do that for them to share the assets that you mentioned?

Reply:
sort of - you can share assets/caches across workflows, but we need to ensure that some jobs run after others and load the cached version that was generated _from that run_, so it's more an issue of scheduling than of sharing